### ISSUE TITLE
uboot does not follow armbianEnv file

### ISSUE THREAD

**Issue by Jackbeef** [HelmSecure/armbian-images#27]
v2a with 16 G mmc : Trying to boot NVME and have valid gpt for the NVMe drive.  However when rootdev=UUID= is changed in armbianEnv, I see that uboot is not following the directive.
Is there a way to have access to the initial uboot SPI. or have a uboot that is pointed to /dev/nvme* ?
(looking into jtag for further debug)

On v2a with 8 G mmc I was able to boot from nvme (both boot and / )

Thank you.

**Comment by dsigurds** [comment]
Are both of your v2a's running the same software build?  The size of the mmc shouldn't have any impact on being able to boot from nvme.

**Comment by Jackbeef** [comment]
As you suggested the diff in mmc should not matter. One time I was able to use the nand-sata to boot from NVME but that was it. (never was able to replicate again). For now I am booting mmc and point to encrypted nvme  for emails.
Can you check the u-boot and see if can be changed to allow NVME boot? . Also looking at the SOM, I do not see SPI flash ... am I missing something here?  
Thanks.

**Comment by Nec83** [comment]
Noticed similar issue as Jackbeef, can setup the Helm v2a to boot up successfully fine from the eMMC but when trying to redo the whole setup and change to boot from NVMe, the device just doesn't come back up nor receive a reserved IP from the DHCP server. 

Something else I picked up on: Directly after reflashing the Helm with the custom image 48, there's no "armbian-config" available (even after apt update / apt upgrade - attached screenie) but it does find the 2 option "armbian-install" (attached screenie after changes made). Confirmed this does update the armbianEnv.txt file correctly (attached - screenie with the extra args) with all the details and the NVMe UUID, but doing a reboot afterwards causes the device to never come back up.

Is there something I'm just missing completely or need to do extra to get it to boot successfully from the NVMe drive?

Thanks!


![armbianEnv txt](https://user-images.githubusercontent.com/67586800/235335743-3abb6430-633a-4646-b4fb-618f884c6949.png)
![armbian-install-success](https://user-images.githubusercontent.com/67586800/235335745-a69836ee-f93a-4d42-a46c-260b35fdab07.png)
![update-upgrade](https://user-images.githubusercontent.com/67586800/235335746-5b11bf34-bb52-4ba3-9be1-9653db49e1d4.png)

**Comment by Jackbeef** [comment]
Just do apt install armbian-config and it will install it.
I changed my mind and boot mmc now. Nvme is encrypted and if someone wants to get it  ... well has to do some work to get there.
Works stable as email server, I wish we could be uptodate with  working new kernel.

**Comment by Nec83** [comment]
I did the same and happily booting off the MMC... just a bit of a pain knowing that if something goes wrong with the MMC, there's no option to use the NVMe storage option at all.

I did the armbian-config install eventually but that only streamlined some options like renaming the hostname or installing Docker from the "Softy" menu, nothing substantially changed in the bootenv menu unfortunately.

Might just have to use it booting off the MMC for the time being until another option becomes available...

**Comment by Jackbeef** [comment]
I do not think that this so called "project" is going anywhere.  Reason why I purchased this unit was originally because Giri "promised"  to open source it if company goes down south. Well here is for his so called promise.   This will drag until no one is interested or people forget.

**Comment by naturistwiki** [comment]
I bought my v1 unit and was an early supporter based on the same assumption that it would be open sourced and I could continue to use the hardware.  Now I am loosing hope and assume I should start looking into securely wiping it so I can get rid of it as a waste of money.

**Comment by git529** [comment]
Hi. I am having the same issue where I can flash my v2a with the custom armbian .img, then ssh and use armbian-install or armbian-config to install system to nvme0n1p1 (deleted existing partition and created new one with fdisk) but after reboot the Helm's LED stays on and I can't ssh into it anymore.

ANY help would be greatly appreciated!

Thanks.
